enhancements/update/cvo-log-level-api.md

```diff
@@ -0,0 +1,327 @@
+
+In the future, if there is a desire and a need for this feature to function in 
+the HyperShift as well, additional implementation may be done. 
+
+#### Standalone Clusters
+
+In standalone clusters, cluster administrators will be able to specify the desired
+CVO log level. No additional changes are needed.
+
+#### Single-node Deployments or MicroShift
+
+Same as standalone clusters.
+
+### Implementation Details/Notes/Constraints
+
+This enhancement proposes to create a new `operator/v1/types_clusterversion.go` 
+file in the [OpenShift API repository](https://github.com/openshift/api). 
+Meaning the new data types will be part of the [`github.com/openshift/api/operator/v1`][github.com/openshift/api/operator/v1]
+package. A new feature-gated cluster scoped configuration API resource will be 
+defined to configure the CVO.
+
+The types are inspired by the existing [`OperatorSpec`][OperatorSpec]
+and [`OperatorStatus`][OperatorStatus]
+structures and copy over some of the fields. Only the  
+`OperatorLogLevel` field is introduced to the `spec` field as there is currently
+no need to support the rest of the [`OperatorSpec`][OperatorSpec]
+fields or any new fields. A similar case applies to the 
+[`OperatorStatus`][OperatorStatus] fields. The `ObservedGeneration` field is 
+used to give feedback regarding the last generation change the CVO has dealt
+with. There is currently no need to support additional fields.
+
+The `ClusterVersionOperator` CRD will be behind a new 
+`ClusterVersionOperatorConfiguration` FeatureGate that will be used to 
+control the development of the CVO configuration.
+
+[github.com/openshift/api/operator/v1]: https://github.com/openshift/api/tree/master/operator/v1
+[OperatorSpec]: https://github.com/openshift/api/blob/f89ab92f1597eaed4de5b947c1781adde2bf42fb/operator/v1/types.go#L54
+[OperatorStatus]: https://github.com/openshift/api/blob/f89ab92f1597eaed4de5b947c1781adde2bf42fb/operator/v1/types.go#L112
+
+The proposed contents of the new `operator/v1/types_clusterversion.go` file:
+
+```go
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterVersionOperator holds cluster-wide information about the Cluster Version Operator.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +openshift:file-pattern=cvoRunLevel=0000_00,operatorName=cluster-version-operator,operatorOrdering=01
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=clusterversionoperators,scope=Cluster
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2044
+// +openshift:enable:FeatureGate=ClusterVersionOperatorConfiguration
+type ClusterVersionOperator struct {
```

Comment:
You should be able to add CEL at this level to prevent the `metadata.name` being anything other than `cluster`, which saves you having to work out what would happen if a user created one that was not called `cluster`.

(Though this is probably a worthy discussion point you could touch on in the doc)

Reply:
Oh, neat, thanks! I did not think of that.

Reply:
I have updated the enhancement, thanks. CEL is being proposed to enforce only one `cluster` `ClusterVersionOperator` CR.